pkg/libzmix/bbs/bbs.go

```diff
@@ -0,0 +1,709 @@
+}
+
+//SetPublicKey sets the bbs public key on the BlindCommitmentContext
+func (r *BlindCommitmentContext) SetPublicKey(publicKey HandleByteBuffer) error {
+	var err C.ExternError
+
+	code := C.bbs_blind_commitment_context_set_public_key(r.data, publicKey.Buffer, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred setting secret key to BBS sign context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//AddMessage adds a message and its associated index to the BlindCommitmentContext
+func (r *BlindCommitmentContext) AddMessage(message string, index int) error {
+	var err C.ExternError
+	var cMessage C.ByteBuffer
+	defer C.bbs_byte_buffer_free(cMessage)
+
+	cMessage.len = (C.long)(len(message))
+	cm := C.CString(message)
+	cMessage.data = (*C.uchar)(unsafe.Pointer(cm))
+	code := C.bbs_blind_commitment_context_add_message_bytes(r.data, (C.uint)(index), cMessage, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred adding message bytes to BBS blind commitment context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+//SetNonce sets the nonce on the BlindedCommitmentContext
+func (r *BlindCommitmentContext) SetNonce(nonce []byte) error {
+	var err C.ExternError
+	var nonceBuffer C.ByteBuffer
+	defer C.bbs_byte_buffer_free(nonceBuffer)
+
+	nonceBuffer.len = (C.long)(len(nonce))
+	nonceBuffer.data = (*C.uchar)(unsafe.Pointer(&nonce))
+
+	code := C.bbs_blind_commitment_context_set_nonce_bytes(r.data, nonceBuffer, &err)
+	if code != 0 {
+		return errors.Errorf("error occurred setting nonce bytes on BBS blind commitment context: %s", C.GoString(err.message))
+	}
+
+	return nil
+}
+
+type BlindedCommitment struct {
+	Commitment     C.ByteBuffer
+	Context        C.ByteBuffer
+	BlindingFactor C.ByteBuffer
+}
+
+//Finish generates a commitment, context, and blinding factor from the BlindCommitmentContext
+func (r *BlindCommitmentContext) Finish() (BlindedCommitment, error) {
+	var err C.ExternError
+	var commitment C.ByteBuffer
+	var outContext C.ByteBuffer
+	var blindingFactor C.ByteBuffer
+
+	code := C.bbs_blind_commitment_context_finish(r.data, &commitment, &outContext, &blindingFactor, &err)
+	if code > 0 {
+		return BlindedCommitment{}, errors.Errorf("error occurred finishing BBS sign context: %s", C.GoString(err.message))
+	}
+
+	return BlindedCommitment{
+		Commitment:     commitment,
+		Context:        outContext,
+		BlindingFactor: blindingFactor,
+	}, nil
+}
+
+//FreeCommitment deallocates the Commitment instance from the BlindedCommitment
+func (r *BlindedCommitment) FreeCommitment() {
+	C.bbs_byte_buffer_free(r.Commitment)
+}
+
+//FreeContext deallocates the Context instance from the BlindedCommitment
+func (r *BlindedCommitment) FreeContext() {
+	C.bbs_byte_buffer_free(r.Context)
+}
+
+//FreeBlindingFactor deallocates the BlindingFactor instance from the BlindedCommitment
+func (r *BlindedCommitment) FreeBlindingFactor() {
+	C.bbs_byte_buffer_free(r.BlindingFactor)
+}
+
```

Comment:
I think it would make sense here to combine all these into one `Free` method on `BlindCommitment`.